package.json

```diff
@@ -687,6 +693,13 @@
                     "when": "resourceFilename==function.json",
                     "group": "zzz_binding@1"
                 }
+            ],
+            "editor/title": [
+                {
+                    "command": "azureFunctions.eventGrid.sendMockRequest",
+                    "when": "resourceFilename=~/.*.eventgrid.json$/",
```

Comment:
> Because of the way the button and code lens need to be registered at the extension level, they appear in any .eventgrid.json file, even if no function is currently being executed or the function app is even running. Is there a way to stop them from showing unless the function app is running and a function is being executed?

Regarding this question: I think there are a couple of ways we can handle this.

1. You should be able to define a extension context value that tracks whether or not the function host is running. Then, you would use that context value and check for it in this `when` clause to only show the button when an app is active.
2. When we execute the trigger, part of the implementation of execute should verify if the local host is running. If it isn't, then we can start it for the user and then execute the trigger.

Reply:
1. How do I check for a context value in the `when` clause?
2. I already do that, but if the host isn't running, I simply just throw an error. The challenge of starting it up for them is which function exactly should I run? In the example that I showed, there was only one eventgrid function in the entire function app, but what if there were multiple? So not only do we need the host to be running, but also a specific function should be in the process of being executed I think.

Reply:
Discussed offline. Changed it such that the file stays open after a request is sent, and the button sends the request to the last function that was executed, until the file is closed or another function is executed.